File: Python Labs/com/pattern/builder/ClothingItemBuilder.py
from com.pattern.builder.ClothingItem import ClothingItem


class ClothingItemBuilder:
    def __init__(self):
        self.type = None
        self.color = None
        self.size = None

    def set_type(self, type):
        self.type = type
        return self

    def set_color(self, color):
        self.color = color
        return self

    def set_size(self, size):
        self.size = size
        return self

    def build(self):
        return ClothingItem(self.type, self.color, self.size)

File: Java Labs/Patterns On Java/src/com/pattern/builder/ClothingItemBuilder.java
package com.pattern.builder;

public class ClothingItemBuilder {
    private String type;
    private String color;
    private int size;

    public ClothingItemBuilder setType(String type) {
        this.type = type;
        return this;
    }

    public ClothingItemBuilder setColor(String color) {
        this.color = color;
        return this;
    }

    public ClothingItemBuilder setSize(int size) {
        this.size = size;
        return this;
    }

    public ClothingItem build() {
        return new ClothingItem(type, color, size);
    }
}
